### ISSUE TITLE
[CA2225] Alternative name for type conversion operators is nonsensical

### ISSUE THREAD

**Issue by Echo-8-ERA** [dotnet/roslyn-analyzers#1717]
#### Analyzer package
[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version
v2.6.1

#### Diagnostic ID
[CA2225](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2225-operator-overloads-have-named-alternates)

#### Repro steps
```c#
public class Bar {
	
	public int i {get; set; }

	public Bar(int i) => this.i = i;
	
}

public class Foo {
	
	public int i {get; set; }

	public Foo(int i) => this.i = i;

	public static implicit operator Foo(Bar b) => new Foo(b.i);

	public static Foo FromBar(Bar b) => new Foo(b.i);

}
```
#### Expected behavior
No violation of CA2225.

#### Actual behavior
CA2225 violation. If `FromBar` is renamed to `FromFoo`, it goes away, but then that name is the complete opposite of what the method is doing.
